### ISSUE TITLE
LD_LIBRARY_PATH mod causes trouble for forked system applications

### ISSUE THREAD

**Issue by ThomasWaldmann** [pyinstaller/pyinstaller#1759]
See there https://github.com/borgbackup/borg/pull/538 and there https://github.com/borgbackup/borg/issues/514.

Not sure there can be done anything against that issue in pyinstaller except maybe documenting that there is a problem and offering some easy way for the application to find out whether it is executing under pyinstaller and what was added to LD_LIBRARY_PATH by pyinstaller (it is not said that this path is otherwise empty) without it having to make guesses.

BTW, at the place where this gets done in pyinstaller, there is an interesting comment about that this would be troublesome on OS X. Well, seems it can also cause trouble on other POSIX OSes (as seen above).

**Comment by dimaqq** [comment]
Yes, it deserves to be documented (at least)

I've had to monkey-patch `subprocess.Popen` to always pass `env=` value being `os.environ` sans `LD_LIBRARY_PATH`.

I think it should be possible to use `-Wl,rpath` instead of `LD_LIBRARY_PATH`

**Comment by tallforasmurf** [comment]
Would this be an appropriate entry for the "Recipes" section (https://github.com/pyinstaller/pyinstaller/wiki/Recipes) ? It sounds like a similar kind of corner case to those others.

If it really is general enough to be in the manual, somebody please ELI5 what is actually going on in detail, I'll take a stab at it.

**Comment by ThomasWaldmann** [comment]
See also #625.

**Comment by ThomasWaldmann** [comment]
The latest incarnation of this troublemaker is that fakeroot somehow does not work with the single-file binary for borgbackup git master. It works for 1.0-maint branch though.

fakeroot sets LD_PRELOAD and LD_LIBRARY_PATH so it finds its own stuff.

**Comment by PlasmaPower** [comment]
If anyone wants exact replication instructions for the fakeroot issue with borg: `curl -L https://gist.githubusercontent.com/PlasmaPower/4b0fef40059685e81b8d3fcea816832f/raw/ca3fa110090b2941532e2b31944deea4e2727401/wheezy32-test.log` (use curl because it has terminal escape codes, captured via `script`)

**Comment by ThomasWaldmann** [comment]
I made PR #2148 to fix this, please review.

**Comment by ThomasWaldmann** [comment]
Fixed by #2148.

Applications that use pyinstaller and want to make sure a forked system application uses the original value of LD_LIBRARY_PATH can set LD_LIBRARY_PATH=$LD_LIBRARY_PATH_ORIG in the environment given to subprocess.Popen.

Hmm, add this to the docs? Where?

**Comment by htgoebel** [comment]
> Hmm, add this to the docs? Where?

runtime-information.rst seems to be a good place.
